internal/controllers/module_reconciler.go

```diff
@@ -103,6 +108,17 @@ func (mr *ModuleReconciler) Reconcile(ctx context.Context, mod *kmmv1beta1.Modul
 		return ctrl.Result{}, fmt.Errorf("failed to get list of nodes by selector: %v", err)
 	}
 
+	images, err := mr.reconHelper.prepareImages(ctx, mod, targetedNodes)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to prepare moduleImagesConfig's images for module %s/%s: %v",
+			mod.Namespace, mod.Name, err)
+	}
+
+	if err := mr.micAPI.ApplyMIC(ctx, mod.Name, mod.Namespace, images, mod.Spec.ImageRepoSecret, mod); err != nil {
```

Comment:
why not make ApplyMic part of prepareImage, and rename prepare Image into something like: handleMICImages

Reply:
Fixed.